python/triton/language/standard.py

```diff
@@ -426,7 +441,7 @@ def flip(x, dim=None):
         for j in core.static_range(0, steps + 1):
             if j != i and j != i + 1:
                 flip2 = core.expand_dims(flip2, j)
-        y = sum(y * flip2, i + 1, keep_dims=True)
+        y = sum(y * flip2, i + 1, keep_dims=True, dtype=y.dtype)
     x = core.reshape(y, x.shape).to(x.dtype, bitcast=True)
```

Comment:
I did this to maintain the existing behaviour, but not sure if it would be better drop the `bitcast=True` instead.

Reply:
I believe it is still needed? It is there for cases where we have floats, we want to do the sum in the equivalent int format

Reply:
Yes, without the bitcast we could have `0 * inf = nan`. Though I wonder if this should be updated to use `tl.gather`?

Reply:
Even more, why don't we move this to use `xor_sum` rather than `sum`?

Reply:
I suspect `sum` is better because it can generate multiply-add instructions.

Reply:
If we really care about perf, we'll be able to implement this via a `convert_layout` (once `convert_layout` is fast in general) by performing the `i -> n - i` transformation on the bases along the flipped dimension and converting to this new layout.

Reply:
Isn't that layout non-linear since the 0th element is not at the 0-index?

Reply:
Yes, sorry, for this we would need an affine map, in particular the map i -> i ^ (shape[d]-1) (where ^ is the xor).

Reply:
> Yes, without the bitcast we could have 0 * inf = nan. Though I wonder if this should be updated to use tl.gather?

I do think we should just go use tl.gather